Scripts/ParseEmailFiles/ParseEmailFiles.py

```diff
@@ -3241,13 +3241,19 @@ def get_utf_string(text, field):
 def convert_to_unicode(s):
     global ENCODINGS_TYPES
     try:
-        encoded_s, encoding = decode_header(s)[0]
-        s = encoded_s.decode(encoding).encode('utf-8').strip()
-        ENCODINGS_TYPES.add(encoding)
-    except Exception:
+        res = ''  # utf encoded result
+        for decoded_s, encoding in decode_header(s):  # return a list of pairs(decoded, charset)
+            if encoding:
+                res += decoded_s.decode(encoding).encode('utf-8')
+                ENCODINGS_TYPES.add(encoding)
+            else:
+                res += decoded_s
+        return res.strip()
+    except Exception as ex:
```

Comment:
shouldn't we catch only UnicodeDecodeError exceptions here?

Reply:
There are different errors that can be raised. We want to catch them all (this is also how it was before my fix) as we don't want to fail the email parsing because of a faulty header. At most we return the original header without decoding.